model/task/db.go

```diff
@@ -2839,6 +2840,16 @@ func SetHasAnnotations(taskId string, execution int) error {
 	return errors.Wrapf(err, "marking task '%s' as having annotations", taskId)
 }
 
+// SetNumNextTaskDispatches sets the next task count for a task.
+func SetNumNextTaskDispatches(taskId string, execution int, count int) error {
+	err := UpdateOne(
+		ByIdAndExecution(taskId, execution),
+		bson.M{"$set": bson.M{
```

Comment:
Should this be `$inc` rather than `$set` with an incremented count? `$inc` is slightly safer because it'll increment the counter in a way that's thread-safe.